### PR TITLE
UINV-154 Add separate permissions for transactions

### PR DESCRIPTION
**Pull request by AndreiShumski2**
## Purpose
https://issues.folio.org/browse/UINV-154

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [x] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -124,6 +124,24 @@
         "subPermissions": [
           "finance.acquisitions-units-assignments.manage"
         ]
+      },
+      {
+        "permissionName": "ui-finance.transactions.allocations.create",
```
Comment (aliaksei-chumakou) [inline package.json:129]: IMHO, `transactions` is not needed here, `ui-finance.allocations.create` makes the work.
But it doesn't solve the ticket at all, user will have all the permissions still, `ui-finance.all`. So it should be revised as well

File: translations/ui-finance/en.json [hunk 2]
```diff
@@ -521,5 +521,8 @@
   "fund.name.isInUse": "This Fund name is already in use on the selected Ledger.",
   "fund.code.isInUse": "This Fund code is already in use.",
   "fund.actions.nameIsNotUnique.confirmation.heading": "Save fund",
-  "fund.actions.nameIsNotUnique.confirmation.message": "Warning: You are about to add a fund with the same name as an existing fund. Would you like to complete this action?"
+  "fund.actions.nameIsNotUnique.confirmation.message": "Warning: You are about to add a fund with the same name as an existing fund. Would you like to complete this action?",
+
+  "permission.ui-finance.transactions.allocations.create": "Finance: Create allocations",
```
Comment (aliaksei-chumakou) [inline translations/ui-finance/en.json:526]: in ui-orders you've added translation using another template, which one is correct?
https://github.com/folio-org/ui-orders/blob/master/translations/ui-orders/en.json#L878

Comment (AndreiShumski2) [reply translations/ui-finance/en.json]: in orders. fixed

File: package.json (hunk 1 above)
Comment (AndreiShumski2) [reply package.json]: Ok. renamed it. Regarding `ui-finance.all` let's wait for PO's answer

Comment (AndreiShumski2) [reply package.json]: `ui-finance.all` permissions were updated based on Dennis [coment](https://issues.folio.org/browse/UINV-154?focusedCommentId=78523&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-ortabpanel#comment-78523) and [mod-finance module descriptor](https://github.com/folio-org/mod-finance/blob/master/descriptors/ModuleDescriptor-template.json#L889)

File: package.json [hunk 3]
```diff
@@ -103,7 +103,25 @@
         "description": "",
         "visible": true,
         "subPermissions": [
-          "finance.all",
+          "finance.awaiting-payment.item.post",
```
Comment (aliaksei-chumakou) [inline package.json:106]: I don't think we post anything to it from UI, according to https://wiki.folio.org/display/FOLIJET/Proposed+Permission+Set+Guidelines#ProposedPermissionSetGuidelines-Using*.allPermissions
_use just the permissions actually needed_

File: package.json [hunk 4]
```diff
@@ -103,7 +103,25 @@
...
+          "finance.budgets.all",
+          "finance.credits.item.post",
```
Comment (aliaksei-chumakou) [inline package.json:108]: the same

File: package.json [hunk 5]
```diff
@@ -103,7 +103,25 @@
...
+          "finance.fund-types.all",
+          "finance.funds.all",
+          "finance.group-fiscal-year-summaries.collection.get",
+          "finance.group-fund-fiscal-years.all",
+          "finance.groups.all",
+          "finance.invoice-transaction-summaries.item.post",
+          "finance.ledgers.all",
+          "finance.order-transaction-summaries.item.post",
+          "finance.order-transaction-summaries.item.put",
+          "finance.payments.item.post",
```
Comment (aliaksei-chumakou) [inline package.json:121]: the same

File: package.json [hunk 6]
```diff
@@ -103,7 +103,25 @@
...
+          "finance.release-encumbrance.item.post",
```
Comment (aliaksei-chumakou) [inline package.json:122]: the same

File: package.json (hunk 3 above)
Comment (AndreiShumski2) [reply package.json]: Correct, removed them

**Pull request merged**